packages/taquito-signer/src/helpers.ts

```diff
@@ -0,0 +1,43 @@
+import { b58cencode, prefix } from "@taquito/utils";
+import { PrivateKey as PrivateKeyEd } from "./derivation-tools/ed25519";
+import { PrivateKey as PrivateKeyEc } from "./derivation-tools/ecdsa";
+import { Path } from "./derivation-tools";
+import { InvalidCurveError, ToBeImplemented } from "./errors";
+
+export type Curves = 'ed25519' | 'secp256k1' | 'p256' | 'bip25519';
+
+// bip32 when supported add to @param curve bip25519
+/**
+ *
+ * @param seed bip39.mnemonicToSeed
+ * @param derivationPath Tezos Requirement 44'/1729' for HD key address default 44'/1729'/0'/0'
+ * @param curve 'ed25519' | 'secp256k1' | 'p256''
+ * @returns final Derivation of HD keys tezos Secret key
+ */
+export const generateSecretKey = (seed: Uint8Array, derivationPath: string, curve: Curves) => {
+  const path = Path.fromString(derivationPath);
+  let node: PrivateKeyEc | PrivateKeyEd;
+
+  switch (curve) {
+    case 'ed25519': {
+      node = PrivateKeyEd.fromSeed(seed).derivePath(path);
+      const sk = b58cencode(node.seed().slice(0, 32), prefix.edsk2);
+      return sk;
+    }
+    case 'secp256k1':
```

Comment:
if this curve isn't implemented, we should throw an error/exception clearly as to not confuse the users

Reply:
It doesn't mean that secp256k1 is not implemented; if the curve is `secp256k1` or `p256`, we enter the `p256` case.

Reply:
my mistake, that's not an issue then